packages/core/src/locales/pt.ts

```diff
@@ -0,0 +1,123 @@
+import type { $ZodStringFormats } from "../checks.js";
+import type * as errors from "../errors.js";
+import * as util from "../util.js";
+
+const Sizable: Record<string, { unit: string; verb: string }> = {
+  string: { unit: "caracteres", verb: "ter" },
+  file: { unit: "bytes", verb: "ter" },
+  array: { unit: "elementos", verb: "ter" },
+  set: { unit: "elementos", verb: "ter" },
+};
+
+function getSizing(origin: string): { unit: string; verb: string } | null {
+  return Sizable[origin] ?? null;
+}
+
+export const parsedType = (data: any): string => {
+  const t = typeof data;
+
+  switch (t) {
+    case "number": {
+      return Number.isNaN(data) ? "NaN" : "número";
+    }
+    case "object": {
+      if (Array.isArray(data)) {
+        return "array";
+      }
+      if (data === null) {
+        return "nulo";
+      }
+      if (Object.getPrototypeOf(data) !== Object.prototype) {
+        return data.constructor.name;
+      }
+    }
+  }
+  return t;
+};
+
+const Nouns: {
+  [k in $ZodStringFormats | (string & {})]?: string;
+} = {
+  regex: "entrada",
+  email: "endereço de e-mail",
+  url: "URL",
+  emoji: "emoji",
+  uuid: "UUID",
+  uuidv4: "UUIDv4",
+  uuidv6: "UUIDv6",
+  nanoid: "nanoid",
+  guid: "GUID",
+  cuid: "cuid",
+  cuid2: "cuid2",
+  ulid: "ULID",
+  xid: "XID",
+  ksuid: "KSUID",
+  datetime: "data e hora ISO",
+  date: "data ISO",
+  time: "hora ISO",
+  duration: "duração ISO",
+  ipv4: "endereço IPv4",
+  ipv6: "endereço IPv6",
+  cidrv4: "faixa IPv4",
+  cidrv6: "faixa IPv6",
+  base64: "string codificada em base64",
```

Comment:
`string` can be `texto` instead. It's better to get a full translation into Portuguese, do you agree?